main.go

```diff
@@ -80,43 +79,49 @@ func run() error {
 		return err
 	}
 
-	if flag.NArg() > 1 {
-		return errors.New("must specify exactly one path argument (or zero for stdin)")
-	}
-
-	root, err := filepath.EvalSymlinks(flag.Arg(0))
-	if err != nil {
-		return fmt.Errorf("following symlinks in input path: %s", err)
-	}
-
 	var ignoreRE *regexp.Regexp
+	var err error
 	if len(*ignore) > 0 {
 		ignoreRE, err = regexp.Compile(*ignore)
 		if err != nil {
 			return fmt.Errorf("compiling ignore regexp: %s", err)
 		}
 	}
 
-	err = filepath.Walk(root, func(path string, fi os.FileInfo, err error) error {
-		if os.IsNotExist(err) {
-			return nil
-		} else if err != nil {
-			return err
-		}
-		if ignoreRE != nil && ignoreRE.MatchString(path) {
-			return nil
-		}
-		if fi.IsDir() {
-			return nil
+	visited := make(map[string]interface{})
```

Comment:
A nit. `map[string]bool{}` or `map[string]struct{}` would be clearer here.

Reply:
oops yes. I meant to do the struct{} thing.  will fix.